### PR TITLE
PSP-5492 Update Historical # and Type on Property Details (Property Attributes)

### PR DESCRIPTION
**Pull request by asanchezr**
image

### CODE REVIEW

File: source/backend/api/Areas/HistoricalNumber/HistoricalNumberController.cs [hunk 1]
```diff
@@ -48,13 +52,44 @@ public HistoricalNumberController(IPropertyService propertyService, IMapper mapp
         [HttpGet("{propertyId}/historicalNumbers")]
         [HasPermission(Permissions.PropertyView)]
         [Produces("application/json")]
-        [ProducesResponseType(typeof(HistoricalNumberModel), 200)]
+        [ProducesResponseType(typeof(IEnumerable<HistoricalNumberModel>), 200)]
         [SwaggerOperation(Tags = new[] { "property" })]
         [TypeFilter(typeof(NullJsonResultFilter))]
         public IActionResult GetHistoricalNumbersForPropertyId(long propertyId)
         {
+            _logger.LogInformation(
```
Comment (devinleighsmith) [inline source/backend/api/Areas/HistoricalNumber/HistoricalNumberController.cs:60]: thanks for adding this.

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/UpdateHistoricalNumbersSubForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        <>
+          {values.historicalNumbers?.map((hn, index) => (
+            <React.Fragment key={`property-historical-${index}`}>
+              <Row className="py-3" data-testid={`historicalNumberRow[${index}]`}>
+                <Col xs="auto" xl="5">
+                  <Input field={`historicalNumbers.${index}.historicalNumber`} />
+                </Col>
+                <Col xs="auto" xl="5" className="pl-0">
+                  <Select
+                    data-testid="select-historical-type"
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/UpdateHistoricalNumbersSubForm.tsx:41]: this is the format we generally use for data-testids. (lowercase, hyphenated). would be nice to update line 35 to this format.

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/validation.ts [hunk 3]
```diff
@@ -35,4 +37,17 @@ export const UpdatePropertyDetailsYupSchema = Yup.object().shape({
     postal: Yup.string().max(20, 'Postal must be at most 20 characters'),
   }),
   generalLocation: Yup.string().max(2000, 'General location must be less than 2000 characters'),
+  historicalNumbers: Yup.array().of(
+    Yup.object().shape({
+      historicalNumber: Yup.string(),
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/validation.ts:42]: this should have a character limit of 500 characters.

Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/validation.ts:42]: Will add

**Comment by devinleighsmith** [comment]
When saving an empty row (see screenshot) The database throws an exception:

SqlException: *** [PIMS_FILNUM_I_S_I_TR], Line 8. Errno 515: Cannot insert the value NULL into column 'FILE_NUMBER_TYPE_CODE', table 'pims.dbo.PIMS_FILE_NUMBER'; column does not allow nulls. INSERT fails.
The transaction ended in the trigger. The batch has been aborted.

image

**Comment by devinleighsmith** [comment]
> When saving an empty row (see screenshot) The database throws an exception:
> 
> SqlException: *** [PIMS_FILNUM_I_S_I_TR], Line 8. Errno 515: Cannot insert the value NULL into column 'FILE_NUMBER_TYPE_CODE', table 'pims.dbo.PIMS_FILE_NUMBER'; column does not allow nulls. INSERT fails. The transaction ended in the trigger. The batch has been aborted.
> 
> image

our pattern in these instances is to filter out empty rows when sending to the backend.

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**